gips/0066-graph-horizon.md

```diff
@@ -0,0 +1,348 @@
+- The service provider registers itself on the data service for discoverability and any additional setup that might be required.
+- The service provider can now start providing the service to the data consumers, collecting some form of receipt or voucher that allows them to later redeem fees through the protocol.
+
+## The staking protocol
+
+At it’s core, Graph Horizon includes a staking protocol designed to provide economic security for the exchange of data between a data provider and a data consumer. The staking primitives allow for arbitrary data services to be built leveraging the economic security that Horizon mandates. Next, we will describe the most important features of the staking protocol.
+
+### Provisions
+
+The main abstraction that enables composability in the staking protocol are provisions. A ***provision*** is stake assigned to a specific data service. Like many other staking protocols, provisioned stake is locked for an undetermined period of time during which it acts as economic security. Wrongdoings can be punished by slashing the stake and at any point the stake can be thawed to be removed on a later stage from the provision.
+
+A quick side note about terminology: in “*Graph Horizon Explained*” forum post we discussed *allocations* as the general term for stake assigned to a data service. This, however, would cause issues with backwards compatibility as *allocation* is currently a per-subgraph-deployment amount of stake. We now use the term *allocation* for subgraph allocations of stake on the Subgraph Service and *provision* as the term for stake assigned for a specific data service.
+
+When the service provider creates a provision there are a few considerations to make. They need to specify the slashing authority for the provision and the thawing period for withdrawing the provisioned stake. While setting these parameters is entirely under the service provider’s control it’s expected that data services will only accept provisions that fulfill their operational requirements (namely, that the data service contract is the slashing authority and that the thawing period is greater than whatever the data service requires).
+
+It’s important to note that, while similar, provisions are not what the current protocol defines as allocations but rather a generalized version that operates at a higher level. The provision primitive guarantees that a certain amount of service provider stake is locked as economic security to provide a service. How this security is used is up to the data service implementation, the staking protocol offers a set of interfaces to perform common operations like thawing provisioned stake, managing stake delegations, etc.
+
+![provisions](../assets/gip-0066/provisions.png)
```

Comment:
These diagrams were exported without a background, which makes them hard to read if rendered in dark backgrounds (like here on Github if using the dark theme). Maybe consider updating with a white background like some other diagrams found in other sections of this GIP?